src/sidebar/store/util.js

```diff
@@ -66,3 +68,31 @@ export function bindSelectors(namespaces, getState) {
   });
   return boundSelectors;
 }
+
+/**
+ * Return a value from app state when it meets certain criteria.
+ *
+ * `await` returns a Promise which resolves when a selector function,
+ * which reads values from a Redux store, returns non-null.
+ *
+ * @template T
+ * @param {Object} store - Redux store
+ * @param {(s: Store) => T|null} selector - Function which returns a value from the
+ *   store if the criteria is met or `null` otherwise.
+ * @return {Promise<T>}
+ */
+export function awaitStateChange(store, selector) {
```

Comment:
I think this is a sensible move. We could perhaps expose a method on the store that does something similar.